cargo-near-build/src/cargo_native/compile.rs

```diff
@@ -118,6 +104,11 @@ where
     let cargo = std::env::var("CARGO").unwrap_or_else(|_| "cargo".to_string());
     let mut cmd = Command::new(cargo);
 
+    // removing env, which may be implicitly passed when bulding from within a build-script
+    // see https://github.com/near/cargo-near/issues/287
+    // CARGO_ENCODED_RUSTFLAGS="-Awarnings" and RUSTFLAGS="-Awarnings" both result
+    // in mysterious failures of `cargo build --target wasm32-unknown-unknown` (rustc bug)
+    cmd.env_remove("CARGO_ENCODED_RUSTFLAGS");
```

Comment:
The comment says that both `CARGO_ENCODED_RUSTFLAGS` and `RUSTFLAGS` cause problems, but we only remove `CARGO_ENCODED_RUSTFLAGS`, is this intended?

Reply:
RUSTFLAGS gets reset to unset in build scripts, so the 2nd or 3rd of 4th build won't inherit this value from 
previous 1st or 2nd or 3rd one, 
and the 1st build isn't having its value concatenated with value from env as per another comment, 
and will have it defined as `-Awarnings` for abi builds and `-C link-arg=-s` for wasm builds